.pre-commit-config.yaml

```diff
@@ -9,3 +9,21 @@ repos:
       - id: check-json
       - id: check-yaml
       - id: check-added-large-files
+  - repo: http://github.com/astral-sh/ruff-pre-commit
+    rev: v0.12.12
+    hooks:
+      - id: ruff
+        args: [ "--fix", "--show-fixes" ]
+        types_or: [ python, jupyter ]
+        exclude: |
+          (?x)^(
+              gdal_adapter/transform.py|
+              gdal_adapter/reformat.py
```

Comment:
Feels like a cop-out, but I would like to start enforcing formatting standards where we can.